docs/functions.md

```diff
@@ -4,54 +4,143 @@ title: Functions
 sidebar_label: Functions
 ---
 
-[Examples](https://github.com/neon-bindings/examples/tree/legacy/functions)
+Neon's main way of connecting Rust and JavaScript is by allowing you to define **[functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions)** that are implemented in Rust.
 
-## This
+## Defining Functions
 
-In order to call methods on `cx.this`, it must be downcasted to a `JsObject`. For more on [type checking section](type-checking.md#downcasting) for more details
+A Neon function looks and acts to JavaScript like a regular JavaScript function, but its behavior is defined in pure Rust. Defining a Neon function requires first defining a Rust function of type `fn(FunctionContext) -> JsResult<T>` where `T` can be any type that implements the [`Value`](https://docs.rs/neon/latest/neon/types/trait.Value.html) trait:
```

Comment:
This sentence is a little awkward with defining twice. Maybe "creating a Neon function requires first defining"?